lib/benchpark/experiment.py

```diff
@@ -65,30 +114,124 @@ def compute_config_section(self):
+        return []
+
+    def compute_modifiers_section_wrapper(self):
         # by default we use the allocation modifier and no others
-        return [{"name": "allocation"}]
+        modifier_list = [{"name": "allocation"}]
+        modifier_list += self.compute_modifiers_section()
+        for cls in self.helpers:
+            modifier_list += cls.compute_modifiers_section()
+        return modifier_list
+
+    def add_experiment_name_prefix(self, prefix):
+        self.expr_name = [prefix] + self.expr_name 
+
+    def add_experiment_variable(self, name, values, use_in_expr_name=False):
+        self.variables[name] = values
+        if use_in_expr_name:
+            self.expr_name.append(f"{{{name}}}")
+
+    def zip_experiment_variables(self, name, variable_names):
+        self.zips[name] = list(variable_names)
+
+    def matrix_experiment_variables(self, variable_names):
+        if isinstance(variable_names, str):
+            self.matrix.append(variable_names)
+        elif isinstance(variable_names, list):
+            self.matrix.extend(variable_names)
+        else:
+            raise ValueError("Variable list must be of type str or list[str].")
+
+    def add_experiment_exclude(self, exclude_clause):
+        self.excludes.append(exclude_clause)
 
     def compute_applications_section(self):
-        # TODO: is there some reasonable default?
         raise NotImplementedError(
             "Each experiment must implement compute_applications_section"
         )
 
+    def compute_applications_section_wrapper(self):
+        self.expr_name = []
+        self.variables = {}
+        self.zips = {}
+        self.matrix = []
+        self.excludes = []
+
+        self.compute_applications_section()
+
+        expr_name_list = [self.name, self.workload]
+        for cls in self.helpers:
+            helper_prefix = cls.get_helper_name_prefix()
+            if helper_prefix:
+                expr_name_list.append(helper_prefix)
+        expr_name = "_".join(expr_name_list + self.expr_name)
+
+        return {
+            self.name: {
+                "workloads": {
+                    self.workload: {
+                        "experiments": {
+                            expr_name: {
+                                "variants": {"package_manager": "spack"}, 
+                                "variables": self.variables,
+                                "zips": self.zips,
+                                "matrix": self.matrix,
+                                "exclude": {"where" : self.excludes} if self.excludes else {},
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+    def add_spack_spec(self, package_name, spec=None):
+        if spec:
+            self.package_specs[package_name] = {
+                "pkg_spec": spec[0],
+                "compiler": spec[1],
+            }
+        else:
+            self.package_specs[package_name] = {}
+
```

Comment:
@scheibelp @pearce8 Refactored lib/experiment.py to remove much of the boilerplate code from the application experiment.py